pkg/codegen/dotnet/gen_program.go

```diff
@@ -491,6 +496,56 @@ func componentOutputType(pclType model.Type) string {
 	return fmt.Sprintf("Output<%s>", elementType)
 }
 
+type ObjectTypeFromConfigMetadata = struct {
+	TypeName      string
+	ComponentName string
+}
+
+func annotateObjectTypedConfig(componentName string, typeName string, objectType *model.ObjectType) *model.ObjectType {
+	objectType.Annotations = append(objectType.Annotations, &ObjectTypeFromConfigMetadata{
+		TypeName:      typeName,
+		ComponentName: componentName,
+	})
+
+	return objectType
+}
+
+// collectObjectTypedConfigVariables Returns the object types in config variables are need to be generated emitted
+// as classes.
+func collectObjectTypedConfigVariables(component *pcl.Component) map[string]*model.ObjectType {
+	objectTypes := map[string]*model.ObjectType{}
+	for _, config := range component.Program.ConfigVariables() {
+		componentName := Title(component.Name())
+		typeName := configObjectTypeName(config.Name())
```

Comment:
This isn't doing anything for uniqueness? Does it need to, or is that already covered by pcl binding making vars unique? Could these conflict with anything else?

Reply:
> Does it need to

The component program binder will error if you have multiple config variables declared in the same program. This happens at the `declareNode` phase.

> Could these conflict with anything else?

No, they could, _only_ when we start supporting nested object types in config variables but those are not supported yet.